### PR TITLE
chore(amplitude): Rename `3ds-fail` to `fail`

### PR DESCRIPTION
**Pull request by xlisachan**
## Because

- these events applies to all transactions, not just [3DS](https://stripe.com/docs/billing/migration/strong-customer-authentication#what-is) transactions.

## This pull request

- renames `3ds-fail` events as `fail` events
- adds tests for former `3ds-fail` events as well as other missing tests

## Issue that this pull request solves

Closes: [FXA-6959](https://mozilla-hub.atlassian.net/browse/FXA-6959)

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.

[FXA-6959]: https://mozilla-hub.atlassian.net/browse/FXA-6959?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by pdehaan** [comment]
Noice!
```sh
git grep "3ds-fail" packages # no results
git grep -ni "\b\w*3ds\b" packages | cat

packages/fxa-payments-server/src/lib/amplitude.ts:32:  submit3DS: '3ds-submit',
packages/fxa-payments-server/src/lib/amplitude.ts:33:  success3DS: '3ds-success',
packages/fxa-payments-server/src/lib/amplitude.ts:34:  complete3DS: '3ds-complete',
packages/fxa-payments-server/src/lib/amplitude.ts:239:    eventTypeNames.success3DS,
packages/fxa-payments-server/src/lib/amplitude.ts:244:    eventTypeNames.complete3DS,
packages/fxa-payments-server/src/lib/amplitude.ts:264:    eventTypeNames.submit3DS,
packages/fxa-payments-server/src/lib/amplitude.ts:274:    eventTypeNames.success3DS,
packages/fxa-payments-server/src/lib/amplitude.ts:279:    eventTypeNames.complete3DS,
```

File: packages/fxa-payments-server/src/lib/amplitude.ts [hunk 1]
```diff
@@ -31,7 +31,6 @@ const eventTypeNames = {
 
   submit3DS: '3ds-submit',
   success3DS: '3ds-success',
-  fail3DS: '3ds-fail',
```
Comment (biancadanforth) [inline packages/fxa-payments-server/src/lib/amplitude.ts:34]: Oh I see, there's already a `fail: 'fail'` mapping further up.

**Review by biancadanforth — CHANGES_REQUESTED** [review]
Thanks Lisa; I'm surprised the existing tests passed with this change (not a good sign unfortunately). Can you update `apiClient.test.js` to assert that these helpers are `calledOnceWithExactly` instead of `calledOnceWith` to verify it is called with this updated argument?

Edit: Or perhaps better, edit packages/fxa-payments-server/src/lib/amplitude.test.ts? That seems to be where we're asserting this sort of thing currently based on your other PR, #15120 .

**Review by IvoJP — APPROVED** [review]

**Comment by IvoJP** [comment]
per our slack convo: amplitude.test.ts was updated so LGTM

**Review by biancadanforth — APPROVED** [review]
Hey Lisa; thanks for the PR! WFM locally.

One overall thought:
* This updates more than the one event described in the original ticket (`fxa_pay_setup - 3ds_fail`), which is fine with me, but it increases the scope slightly in terms of event renaming, so you would also want to:
  * ~~verify the other event in stage/prod~~[1]
  * update the ticket with a comment to this effect
  * increase the scope of @sardesam 's complementary ticket, FXA-6958, to do the same thing for all success events, such that there are no remaining `*3ds*` events emitted by the payments server.
  * update Meg's ticket to that effect
  * clarify what additional events, that are _actually logged_[1], are also being renamed and reply to [Marnie's reminder Slack post](https://mozilla.slack.com/archives/CG0AJ6E77/p1680743430322999?thread_ts=1678247618.837619&cid=CG0AJ6E77) to this effect, CCing Yeonjoo Yoo, who is the only person to my knowledge who has made dashboards based on any SubPlat events to date.

Can you take care of these follow up tasks? Please let me know if you have any questions.

[1]: Interestingly, it turns out we don't record the other event at all right now when a payment method is updated, so no further verification is required on that front. This is a bug, but it's not a bug for a P1 event and therefore not in scope for the current epic.

**Pull request merged**